mailbox/store/src/test/java/org/apache/james/mailbox/store/AbstractCombinationManagerTest.java

```diff
@@ -123,162 +129,187 @@ public void searchFromMailboxManagerShouldReturnMessagesUsingSetInMailboxesFromM
-        ComposedMessageId messageId = messageManager.appendMessage(new ByteArrayInputStream(SUBJECT_TEST_MAIL), new Date(), session, false, recent);
+        ComposedMessageId messageId = messageManager1.appendMessage(new ByteArrayInputStream(SUBJECT_TEST_MAIL), new Date(), session, false, recent);
 
         messageIdManager.setInMailboxes(messageId.getMessageId(), ImmutableList.of(mailbox1.getMailboxId(), mailbox2.getMailboxId()), session);
 
-        messageManager = testingData.createMessageManager(mailbox2, session);
-        assertThat(messageManager.getMetaData(true, session, FetchGroup.UNSEEN_COUNT).getRecent()).containsOnly(messageId.getUid());
+        assertThat(messageManager2.getMetaData(true, session, FetchGroup.UNSEEN_COUNT).getRecent()).containsOnly(messageId.getUid());
     }
 
     @Test
     public void getMetadataFromMessageManagerShouldReturnNumberOfRecentMessageWhenSetInMailboxesFromMessageIdManager() throws Exception {
         Flags recent = new Flags(Flag.RECENT);
-        MessageId messageId = messageManager.appendMessage(new ByteArrayInputStream(SUBJECT_TEST_MAIL), new Date(), session, false, recent).getMessageId();
+        MessageId messageId = messageManager1.appendMessage(new ByteArrayInputStream(SUBJECT_TEST_MAIL), new Date(), session, false, recent).getMessageId();
 
         messageIdManager.setInMailboxes(messageId, ImmutableList.of(mailbox1.getMailboxId(), mailbox2.getMailboxId()), session);
 
-        messageManager = testingData.createMessageManager(mailbox2, session);
-        assertThat(messageManager.getMetaData(true, session, FetchGroup.FIRST_UNSEEN).countRecent()).isEqualTo(1);
+        assertThat(messageManager2.getMetaData(true, session, FetchGroup.FIRST_UNSEEN).countRecent()).isEqualTo(1);
     }
 
     @Test
     public void getMetadataFromMessageManagerShouldReturnUidNextWhenSetInMailboxesFromMessageIdManager() throws Exception {
         Flags recent = new Flags(Flag.RECENT);
-        ComposedMessageId messageId = messageManager.appendMessage(new ByteArrayInputStream(SUBJECT_TEST_MAIL), new Date(), session, false, recent);
+        ComposedMessageId messageId = messageManager1.appendMessage(new ByteArrayInputStream(SUBJECT_TEST_MAIL), new Date(), session, false, recent);
 
         messageIdManager.setInMailboxes(messageId.getMessageId(), ImmutableList.of(mailbox1.getMailboxId(), mailbox2.getMailboxId()), session);
 
-        assertThat(messageManager.getMetaData(true, session, FetchGroup.FIRST_UNSEEN).getUidNext()).isGreaterThan(messageId.getUid());
+        List<MessageResult> listMessages = messageIdManager.getMessages(ImmutableList.of(messageId.getMessageId()), FetchGroupImpl.MINIMAL, session);
+
+        long uid2 = FluentIterable.from(listMessages)
+            .filter(messageInMailbox2())
+            .get(0)
+            .getUid()
+            .asLong();
+
+        assertThat(messageManager2.getMetaData(true, session, FetchGroup.FIRST_UNSEEN).getUidNext().asLong()).isEqualTo(uid2 + 1);
```

Comment:
isGreaterThan(uid2)

(I can increase uid2 by more than 1, it will still be valid...)